### ISSUE TITLE
Adding an audio unit filter to channel straight after it has been removed crash.

### ISSUE THREAD

**Issue by jmarshallgtl** [TheAmazingAudioEngine/TheAmazingAudioEngine#222]
Hi there, i've noticed that in 1.5.6 if you add an audio unit filter to a channel straight after you have removed filters from the same channel then TAAE crashes in the callback. This does not seem to happen in 1.5.5.
Cheers.

**Comment by michaeltyson** [comment]
Hi! D'oh, I'm sorry to hear that. Would you be able to knock up a quick sample app that demonstrates this? Ideally, a patch against the TAAE sample app would be perfect.

**Comment by jmarshallgtl** [comment]
HI Michael, thanks for getting back.

I've recreated the crash in the sample app. Again it only happens in 1.5.6. I've modified the "reverbSwitchChanged:" method. It now adds three audio unit effects to the _loop1 channel (Drums). First I create an array of effects, then remove any effects which may already be on the channel, then add the array to the channel.

Cheers!

[TheAmazingAudioEngine - Filter Crash.zip](https://github.com/TheAmazingAudioEngine/TheAmazingAudioEngine/files/136043/TheAmazingAudioEngine.-.Filter.Crash.zip)
